root/artist/ArtistIndex.js

```diff
@@ -0,0 +1,322 @@
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import {withCatalystContext} from '../context';
+import Annotation from '../static/scripts/common/components/Annotation';
+import DescriptiveLink from '../static/scripts/common/components/DescriptiveLink';
+import Filter from '../static/scripts/common/components/Filter';
+import {type FilterFormT} from '../static/scripts/common/components/FilterForm';
+import WikipediaExtract
+  from '../static/scripts/common/components/WikipediaExtract';
+import {addColonText} from '../static/scripts/common/i18n/addColon';
+import commaOnlyList, {commaOnlyListText} from '../static/scripts/common/i18n/commaOnlyList';
+import {bracketedText} from '../static/scripts/common/utility/bracketed';
+import FormSubmit from '../components/FormSubmit';
+import RecordingList from '../components/list/RecordingList';
+import ReleaseGroupList from '../components/list/ReleaseGroupList';
+import * as manifest from '../static/manifest';
+import entityHref from '../static/scripts/common/utility/entityHref';
+
+import ArtistLayout from './ArtistLayout';
+
+type RelatedArtistsProps = {|
+  +children: React$Node,
+  +title: string,
+|};
+
+const RelatedArtists = ({children, title}: RelatedArtistsProps) => (
+  <p>
+    <strong>{addColonText(title)}</strong>
+    {' '}
+    {children}
+  </p>
+);
+
+type Props = {|
+  +$c: CatalystContextT,
+  +ajaxFilterFormUrl: string,
+  +artist: ArtistT,
+  +eligibleForCleanup: boolean,
+  +filterForm: ?FilterFormT,
+  +hasFilter: boolean,
+  +includingAllStatuses: boolean,
+  +showingVariousArtistsOnly: boolean,
+  +legalName: ?ArtistT,
+  +legalNameAliases: ?$ReadOnlyArray<string>,
+  +legalNameArtistAliases: ?$ReadOnlyArray<string>,
+  +numberOfRevisions: number,
+  +otherIdentities: $ReadOnlyArray<ArtistT>,
+  +recordings: ?$ReadOnlyArray<RecordingT>,
+  +releaseGroups: ?$ReadOnlyArray<ReleaseGroupT>,
+  +wantAllStatuses: boolean,
+  +wantVariousArtistsOnly: boolean,
+  +wikipediaExtract: WikipediaExtractT,
+|};
+
+const ArtistIndex = ({
+  $c,
+  ajaxFilterFormUrl,
+  artist,
+  eligibleForCleanup,
+  filterForm,
+  hasFilter,
+  includingAllStatuses,
+  legalName,
+  legalNameAliases,
+  legalNameArtistAliases,
+  numberOfRevisions,
+  otherIdentities,
+  recordings,
+  releaseGroups,
+  showingVariousArtistsOnly,
+  wantAllStatuses,
+  wantVariousArtistsOnly,
+  wikipediaExtract,
+}: Props) => {
+  const hasRecordings = !!(recordings && recordings.length);
+  const hasReleaseGroups = !!(releaseGroups && releaseGroups.length);
+  const artistLink = entityHref(artist);
+  let message = '';
+
+  if (hasRecordings) {
+    message = l(
+      'This artist has no release groups, only standalone recordings.',
+    );
+  } else if (!hasReleaseGroups && hasFilter) {
+    message = l('No release groups found.');
```

Comment:
Minor: "No release groups found that match this search" or something might be more clear?

Reply:
Also, releases and recordings don't actually have different wording if hasFilter, which is confusing. I'd suggest adding that there too.